lib/qc_apply.dart

```diff
@@ -0,0 +1,30 @@
+String? compressStr(String? initialStr) {
+  if (initialStr == null) return null;
+
+  if (initialStr == '') return '';
+
+  Map<int, String> _chars =
+      initialStr.replaceAll(RegExp(r'[^a-zA-Z]'), '').split('').asMap();
+
+  String _compressedStr = '';
+
+  int _repeatCount = 1;
+
+  _chars.forEach((idx, e) {
+    if (_chars[idx] == _chars[idx - 1]) {
+      _repeatCount++;
+    } else {
+      if (_repeatCount == 1) {
+        _compressedStr += _chars[idx].toString();
+      } else {
+        _compressedStr += _repeatCount.toString() + _chars[idx].toString();
+        _repeatCount = 1;
+      }
```

Comment:
Getting into a second level of `if`s is often an opportunity to consider other ways. The exercise is described as quick so you did the right thing here by not getting stuck on working through alternatives. 👍